cmplibrary/src/test/java/com/sourcepoint/cmplibrary/SourcePointClientTest.java

```diff
@@ -0,0 +1,205 @@
+
+                if (onSuccess) {
+                    String stringToParse = "{\"gdprApplies\":\"http://google.com\"}";
+                    try {
+                        jsonObject = new JSONObject(stringToParse);
+                    } catch (JSONException err) {
+                        System.out.println("Error " + err.toString());
+                    }
+
+                    Header[] headers = new Header[]{};
+                    listener.onSuccess(200, headers, jsonObject);
+                } else {
+                    Header[] headers = new Header[]{};
+                    listener.onFailure(404, headers, "Error", new ConsentLibException("Error"));
+                }
+
+                return null;
+            }
+        }).when(http).get(anyString(), any(ResponseHandlerInterface.class));
+    }
+
+    @Test
+    public void getGDPRStatus() {
+        doGetGDPRStatus(true);
+
+        sourcePointClient.getGDPRStatus(onLoadComplete);
+
+        verify(onLoadComplete, times(1)).onSuccess(eq("http://google.com"));
+
+        verify(onLoadComplete, never()).onFailure(any(ConsentLibException.class));
+
+        reset(onLoadComplete);
+
+        doGetGDPRStatus(false);
+
+        sourcePointClient.getSiteID(onLoadComplete);
+
+        verify(onLoadComplete, never()).onSuccess(any());
+
+        verify(onLoadComplete, times(1)).onFailure(any(ConsentLibException.class));
+    }
+
+    private void doGetCustomConsents(final boolean onSuccess) {
+        doAnswer(new Answer() {
+            @Override
+            public Object answer(InvocationOnMock invocation) throws JSONException {
+                System.out.println("TTTT : CALLED");
+                JSONObject jsonObject = new JSONObject();
+
+                SourcePointClient.ResponseHandler listener = (SourcePointClient.ResponseHandler) invocation.getArguments()[1];
+                String response = "{\"consentedPurposes\":[{\"_id\":\"5d287f273e5ba6241423f58d\",\"name\":\"Personalisation\"},{\"_id\":\"5d287f273e5ba6241423f58e\",\"name\":\"Essential Cookies\"}],\"consentedVendors\":[{\"_id\":\"5b07836aecb3fe2955eba270\",\"name\":\"Google Ad Manager\",\"vendorType\":\"CUSTOM\"}]}";
+
+                if (onSuccess) {
+                    try {
+                        jsonObject = new JSONObject(response);
+                    } catch (JSONException err) {
+                        System.out.println("Error " + err.toString());
+                    }
+                    Header[] headers = new Header[]{};
+                    listener.onSuccess(200, headers, jsonObject);
+                } else {
+                    Header[] headers = new Header[]{};
+                    listener.onFailure(404, headers, "Error", new ConsentLibException("Error"));
+                }
+
+                return null;
+            }
+        }).when(http).get(anyString(), any(ResponseHandlerInterface.class));
+    }
```

Comment:
It seems that the `doAPICallWithAnswer `, `doGetGDPRStatus `, `doGetCustomConsents` have a lot in common. The only difference is in the response. Maybe we can refactor that into a method and receive the response as a parameter?

Reply:
good suggestion , will refactor all code in single method.